### DISCUSSION TITLE
The auth of openfunction

### DISCUSSION THREAD

**Discussion by jackkkkklee** [OpenFunction/OpenFunction#331] [Q&A] [answered]
Hi,community
As a beginner，from what I know right now, the auth part is rely on the k8s gateway  ,the function entry of openfunction. And it varies from different type of k8s gateway.
Will the authentication mechanism be specially developed later or still depend on k8s gateway?

**Comment by benjaminhuo** [comment, marked as answer]
You're right. We'll start to use K8s gateway API as OpenFunction's gateway starting from v0.7.0, find the proposal [here](https://github.com/OpenFunction/OpenFunction/blob/main/docs/proposals/202207-openfunction-gateway.md) 
The PR has been merged into the master branch.

And contour will be the default gateway implementation. Users can also choose to use istio as another gateway implementation.

I think we'll rely on the K8s gateway api's HTTPS for security concerns. @wrongerror you've anything to say?
